iceoryx_binding_c/include/iceoryx_binding_c/types.h

```diff
@@ -74,32 +34,19 @@
 /// The size and the alignment of all structs are verified by the
 /// binding c integration test iox_types_test
 
-struct iox_ws_storage_t_
+typedef struct
 {
-    // the value of the array size is the result of the following formula:
-    // sizeof(WaitSet) / 8
-    /// @note see iceoryx_binding_c_deployment.h.in for calculation of the size
-    uint64_t do_not_touch_me[CALCULATE_STORAGE_SIZE_FOR_WAITSET(IOX_BUILD_GENERATED_MAX_NUMBER_OF_NOTIFIERS) / 8];
-};
-typedef struct iox_ws_storage_t_ iox_ws_storage_t;
+    static_assert(IOX_BINDING_C_STORAGE_WAIT_SET_ALIGNMENT == 8, "Wrong alignment");
+    uint64_t do_not_touch_me[IOX_BINDING_C_STORAGE_WAIT_SET_SIZE / 8];
+} iox_ws_storage_t;
 
-struct iox_user_trigger_storage_t_
+typedef struct
 {
-    // the value of the array size is the result of the following formula:
-    // sizeof(UserTrigger) / 8
-#if defined(__APPLE__)
-    uint64_t do_not_touch_me[15];
-#elif defined(_WIN32)
-    uint64_t do_not_touch_me[18];
-#elif defined(__FreeBSD__)
-    uint64_t do_not_touch_me[8];
-#elif defined(__linux__)
-    uint64_t do_not_touch_me[12];
-#endif
-};
-typedef struct iox_user_trigger_storage_t_ iox_user_trigger_storage_t;
+    static_assert(IOX_BINDING_C_STORAGE_USER_TRIGGER_ALIGNMENT == 8, "Wrong alignment");
```

Comment:
@elBoberido A lower power of two is also fine I think, i.e. 4, 2 or 1. In general everything lower that divides 8 (but alignments are required to be powers of two so ...)

Reply:
@elfenpiff the problem is that the size is a multiple of the alignment and therefore `12 / 8 -> 1` would leave us with a storage which is too small. This could be fixed by `12/8 + (12%8 ? 1 : 0)` but all of our types are 8 bytes aligned and why makes things harder than necessary.